packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -806,6 +829,26 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     if (_snackBars.isNotEmpty)
       _addIfNonNull(children, _snackBars.first._widget, _ScaffoldSlot.snackBar);
 
+    if (config.persistentFooterButtons != null) {
```

Comment:
I suppose you could also check for `config.persistentFooterButtons.length > 0`. On the other hand, maybe it's better to have the substructure in place for the inevitable rebuild with a non-empty buttons list.